source/simulator/solver.cc

```diff
@@ -714,8 +714,8 @@ namespace aspect
     SolverControl solver_control_cheap (parameters.n_cheap_stokes_solver_steps,
                                         solver_tolerance);
 
-    SolverControl solver_control_expensive (system_matrix.block(block_vel,block_p).m() +
-                                            system_matrix.block(block_p,block_vel).m(), solver_tolerance);
+    SolverControl solver_control_expensive (parameters.n_expensive_stokes_solver_steps,
```

Comment:
It has to be a fixed number known at the time of declaring the parameter. So it can't depend on the size of the linear system. But you could use a very large number.

I think 1000 is probably ok. I was just wondering if you thought about this. Mentioning this in the description of the parameter is probably useful.